packages/strapi/lib/middlewares/index.js

```diff
@@ -3,17 +3,19 @@
 const glob = require('glob');
 const path = require('path');
 const { parallel } = require('async');
-const { after, includes, indexOf, drop, dropRight, uniq, defaultsDeep, get, set, isEmpty, isUndefined, union, keys } = require('lodash');
+const { after, includes, indexOf, drop, dropRight, uniq, defaultsDeep, get, set, isEmpty, isUndefined, union } = require('lodash');
 
 module.exports = function() {
   // Set if is admin destination for middleware application.
   // TODO: Use dynamic config for admin url.
   strapi.app.use(async (ctx, next) => {
-    const accepted = keys(strapi.plugins).map(url => `^\/${url}`);
-    accepted.push('^\/admin');
-    const isMatch = accepted.some(rx => new RegExp(rx).test(ctx.request.url));
-    set(ctx, 'admin', isMatch);
-    return next();
+    const accepted = Object.keys(strapi.plugins).map(url => `^\/${url}`).concat(['^\/admin']);
```

Comment:
You should not have nested `strapi.app.use()` and I was saying that the `const accepted = ...` should be out of the middleware.

```js
const accepted = keys(strapi.plugins).map(url => `^\/${url}`);		 +    const accepted = Object.keys(strapi.plugins).map(url => `^\/${url}`).concat(['^\/admin']);

strapi.app.use(async (ctx, next) => {
  ctx.request.admin = accepted.some(rx => new RegExp(rx).test(ctx.request.url));

  return next();
});
```

If the `strapi.plugins` object is not declared at this moment, we should put this middleware somewhere else.